### PR TITLE
Wallet integration

### PR DESCRIPTION
**Pull request by deweller**
Integrates wallet connection via MetaMask

Adds global context to get wallet account throughout the app

Moves wallet button to the top

Allows imports relative to src

### CODE REVIEW

File: frontend/src/components/wallet/WalletConnectionStatus.tsx [hunk 1]
```diff
@@ -0,0 +1,142 @@
+import { useState, useEffect, useContext } from 'react';
+import { utils } from 'ethers';
+import { WalletContext } from 'lib/wallet/WalletContext';
+import metamaskLogo from 'assets/img/metamask.svg';
+
+declare const window: Window &
```
Comment (reisr3) [inline frontend/src/components/wallet/WalletConnectionStatus.tsx:6]: We should probably put this in a `/types` directory in the root of the project.

File: frontend/src/components/wallet/WalletConnectionStatus.tsx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+  typeof globalThis & {
+    ethereum: any;
+  };
+
+interface MetaMaskError {
```
Comment (reisr3) [inline frontend/src/components/wallet/WalletConnectionStatus.tsx:11]: Want to put this in `/interfaces`? Might be useful elsewhere.

File: frontend/src/components/wallet/WalletConnectionStatus.tsx [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+  const shorten = (fullAddress: string) => {
+    return (
+      fullAddress.substring(0, 6) +
+      '...' +
+      fullAddress.substring(fullAddress.length - 4)
+    );
+  };
+
+  return (
+    <div>
```
Comment (reisr3) [inline frontend/src/components/wallet/WalletConnectionStatus.tsx:111]: Need the extra div? Might be able to just do a fragment `<> </>`

**Review by reisr3 — CHANGES_REQUESTED** [review]
Looks awesome! Just a couple things

**Comment by reisr3** [comment]
The error is just a lint error.. I'll work on getting everyone access to Vercel.

```

src/components/wallet/WalletConnectionStatus.tsx
--
14:50:13.001 | Line 20:11:  'walletConnection' is assigned a value but never used  @typescript-eslint/no-unused-vars

```

File: frontend/src/components/wallet/WalletConnectionStatus.tsx (hunk 1 above)
Comment (deweller) [reply frontend/src/components/wallet/WalletConnectionStatus.tsx:6]: I agree about have a central types directory.  This type definition is specific to this one component - it allows checking for `window.ethereum` to check for metamask presence.  I think it makes more sense to leave this one in the same file as the component.

File: frontend/src/components/wallet/WalletConnectionStatus.tsx (hunk 2 above)
Comment (deweller) [reply frontend/src/components/wallet/WalletConnectionStatus.tsx:11]: Agreed.  I moved it and renamed it to IMetaMaskError.

File: frontend/src/components/wallet/WalletConnectionStatus.tsx (hunk 3 above)
Comment (deweller) [reply frontend/src/components/wallet/WalletConnectionStatus.tsx:111]: Nope.  Don't need it.  I fragmentized it.

**Review by reisr3 — APPROVED** [review]
🚀 ⚙️ 🚀 ⚙️ 🚀 ⚙️ 🚀 ⚙️

**Pull request merged**